src/spikeinterface/sortingcomponents/motion/iterative_template.py

```diff
@@ -0,0 +1,292 @@
+     * the 3d histogram is constructed in less cryptic way
+     * peak_locations are computed outside and so can either center fo mass or monopolar trianglation
+       contrary to kilosort2.5 use exclusively center of mass
+
+    See https://www.science.org/doi/abs/10.1126/science.abf4588?cookieSet=1
+
+    Ported by Alessio Buccino into SpikeInterface
+    """
+
+    name = "iterative_template"
+    need_peak_location = True
+    params_doc = """
+    bin_um: float, default: 10
+        Spatial bin size in micrometers
+    hist_margin_um: float, default: 0
+        Margin in um from histogram estimation.
+        Positive margin extrapolate out of the probe the motion.
+        Negative margin crop the motion on the border
+    bin_s: float, default: 2.0
+        Bin duration in second
+    num_amp_bins: int, default: 20
+        number ob bins in the histogram on the log amplitues dimension
+    num_shifts_global: int, default: 15
+        Number of spatial bin shifts to consider for global alignment
+    num_iterations: int, default: 10
+        Number of iterations for global alignment procedure
+    num_shifts_block: int, default: 5
+        Number of spatial bin shifts to consider for non-rigid alignment
+    smoothing_sigma: float, default: 0.5
+        Sigma of gaussian for covariance matrices smoothing
+    kriging_sigma: float,
+        sigma parameter for kriging_kernel function
+    kriging_p: foat
+        p parameter for kriging_kernel function
+    kriging_d: float
+        d parameter for kriging_kernel function
+    """
+
+    @classmethod
+    def run(
+        cls,
+        recording,
+        peaks,
+        peak_locations,
+        direction,
+        rigid,
+        win_shape,
+        win_step_um,
+        win_scale_um,
+        win_margin_um,
+        verbose,
+        progress_bar,
+        extra,
+        bin_um=10.0,
+        hist_margin_um=0.0,
+        bin_s=2.0,
+        num_amp_bins=20,
+        num_shifts_global=15,
+        num_iterations=10,
+        num_shifts_block=5,
+        smoothing_sigma=0.5,
+        kriging_sigma=1,
+        kriging_p=2,
+        kriging_d=2,
+    ):
+
+        dim = ["x", "y", "z"].index(direction)
+        contact_depth = recording.get_channel_locations()[:, dim]
+
+        # spatial histogram bins
+        spatial_bin_edges = get_spatial_bin_edges(recording, direction, hist_margin_um, bin_um)
+        spatial_bin_centers = 0.5 * (spatial_bin_edges[1:] + spatial_bin_edges[:-1])
+
+        # get spatial windows
+        non_rigid_windows, non_rigid_window_centers = get_spatial_windows(
+            contact_depth=contact_depth,
+            spatial_bin_centers=spatial_bin_centers,
+            rigid=rigid,
+            win_margin_um=win_margin_um,
+            win_step_um=win_step_um,
+            win_scale_um=win_scale_um,
+            win_shape=win_shape,
+            zero_threshold=None,
+        )
+
+        # make a 3D histogram
```

Comment:
maybe do a if verbose: print('Computing histograms') as in dredge because otherwise verbose is never used